### PR TITLE
MDEXP-817 - Deleted authority export job profile is displayed for Export all authorities done without specifying profile id on Eureka environment

### PR DESCRIPTION
**Pull request by obozhko-folio**
[MDEXP-817](https://folio-org.atlassian.net/browse/MDEXP-817) - Deleted authority export job profile is displayed for Export all authorities done without specifying profile id on Eureka environment

## Purpose
When export all authorities request doesn’t include profile id in the body - Deleted authority export job profile is displayed as the profile for such export all.

## Approach
Add 'Default' to the query.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] Check logging
  - [x] There are no major code smells or security issues

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/dataexp/repository/JobProfileEntityRepository.java [hunk 1]
```diff
@@ -9,6 +9,6 @@
 
 public interface JobProfileEntityRepository extends JpaRepository<JobProfileEntity, UUID> {
 
-  @Query(value = "SELECT id FROM job_profiles WHERE jsonb ->> 'default' = 'true' AND jsonb ->> 'name' like '%'||:name||'%'", nativeQuery = true)
+  @Query(value = "SELECT id FROM job_profiles WHERE jsonb ->> 'default' = 'true' AND jsonb ->> 'name' like 'Default%'||:name||'%'", nativeQuery = true)
```
Comment (alekGbuz) [inline src/main/java/org/folio/dataexp/repository/JobProfileEntityRepository.java:12]: Hi @obozhko-folio , check please if this logic will be correct if some one will use name parameter starting from 'Default'

Comment (NikitaSedyx) [reply src/main/java/org/folio/dataexp/repository/JobProfileEntityRepository.java:12]: if someone will rename in DB?

Comment (obozhko-folio) [reply src/main/java/org/folio/dataexp/repository/JobProfileEntityRepository.java:12]: In the code only RecordTypeEnum or IdTypeEnum is used as a name parameter, 'Default' is not used.

Comment (NikitaSedyx) [reply src/main/java/org/folio/dataexp/repository/JobProfileEntityRepository.java:12]: I'm happy that smth in code is used, but you check value in DB, not in code

Comment (obozhko-folio) [reply src/main/java/org/folio/dataexp/repository/JobProfileEntityRepository.java:12]: If someone renames it in DB, we will never find it. All job profiles are hardcoded.

Comment (NikitaSedyx) [reply src/main/java/org/folio/dataexp/repository/JobProfileEntityRepository.java:12]: in which places do we hardcode name as well? AFAIK it's the first one where you try to introduce such query

if you don't want to fix base on regular comment, I can request changes to not use hardcode in SQL, try to find other ways to solve it

Comment (obozhko-folio) [reply src/main/java/org/folio/dataexp/repository/JobProfileEntityRepository.java:12]: Here is a folder where all default job profiles are hardcoded: [job profiles](https://github.com/folio-org/mod-data-export/tree/master/src/main/resources/data/job-profiles)

File: src/main/java/org/folio/dataexp/service/DataExportAllService.java [hunk 2]
```diff
@@ -50,8 +50,12 @@ private UUID getJobProfileId(ExportAllRequest exportAllRequest) {
...
   private UUID getDefaultJobProfileId(ExportAllRequest exportAllRequest) {
     if (nonNull(exportAllRequest.getIdType())) {
-      return jobProfileEntityRepository.findIdOfDefaultJobProfileByName(exportAllRequest.getIdType().getValue()).get(0);
+      return switch (exportAllRequest.getIdType()) {
+        case INSTANCE -> UUID.fromString(DEFAULT_INSTANCE_JOB_PROFILE_ID);
+        case HOLDING -> UUID.fromString(DEFAULT_HOLDINGS_JOB_PROFILE_ID);
+        case AUTHORITY -> UUID.fromString(DEFAULT_AUTHORITY_JOB_PROFILE_ID);
+      };
     }
-    return jobProfileEntityRepository.findIdOfDefaultJobProfileByName(ExportRequest.IdTypeEnum.INSTANCE.getValue()).get(0);
```
Comment (alekGbuz) [inline src/main/java/org/folio/dataexp/service/DataExportAllService.java:55]: Hi @obozhko-folio ,  is it possible to remove JobProfileEntityRepository itsef or at least its findIdOfDefaultJobProfileByName method

Comment (obozhko-folio) [reply src/main/java/org/folio/dataexp/service/DataExportAllService.java:55]: Done

**Review by alekGbuz — APPROVED** [review]

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**